app/src/androidTest/java/ch/epfl/sdp/DataForDemo.java

```diff
@@ -218,4 +231,112 @@ public void uploadUserPaths() {
+                    .exceptionally(e -> {
+                        Log.d("BETTER PATH UPLOAD", "Error uploading positions", e);
+                        return null;
+                    });
+            if (infectedOnRoute[i] == 1) {
+                carrierAndPositionCreationUpload(Carrier.InfectionStatus.INFECTED, 1f, lat, lon, timestamp.toDate());
+            }
+            i+=1;
+        }
+        Log.d("LOOPINDEX: ", String.valueOf(i));
+    }
+
+    /**
+     * This method creates a better looking path for demo.
+     * source: https://docs.mapbox.com/android/maps/examples/create-a-line-layer/
+     */
+    private void initRouteCoordinates() {
+        routeCoordinates = new ArrayList<>();
+        routeCoordinates.add(Point.fromLngLat(-118.39439114221236, 33.397676454651766));
+        routeCoordinates.add(Point.fromLngLat(-118.39421054012902, 33.39769799454838));
+        routeCoordinates.add(Point.fromLngLat(-118.39408583869053, 33.39761901490136));
+        routeCoordinates.add(Point.fromLngLat(-118.39388373635917, 33.397328225582285));
+        routeCoordinates.add(Point.fromLngLat(-118.39372033447427, 33.39728514560042));
+        routeCoordinates.add(Point.fromLngLat(-118.3930882271826, 33.39756875508861));
+        routeCoordinates.add(Point.fromLngLat(-118.3928216241072, 33.39759029501192));
+        routeCoordinates.add(Point.fromLngLat(-118.39227981785722, 33.397234885594564));
+        routeCoordinates.add(Point.fromLngLat(-118.392021814881, 33.397005125197666));
+        routeCoordinates.add(Point.fromLngLat(-118.39090810203379, 33.396814854409186));
+        routeCoordinates.add(Point.fromLngLat(-118.39040499623022, 33.39696563506828));
+        routeCoordinates.add(Point.fromLngLat(-118.39005669221234, 33.39703025527067));
+        routeCoordinates.add(Point.fromLngLat(-118.38953208616074, 33.39691896489222));
+        routeCoordinates.add(Point.fromLngLat(-118.38906338075398, 33.39695127501678));
+        routeCoordinates.add(Point.fromLngLat(-118.38891287901787, 33.39686511465794));
+        routeCoordinates.add(Point.fromLngLat(-118.38898167981154, 33.39671074380141));
+        routeCoordinates.add(Point.fromLngLat(-118.38984598978178, 33.396064537239404));
+        routeCoordinates.add(Point.fromLngLat(-118.38983738968255, 33.39582400356976));
+        routeCoordinates.add(Point.fromLngLat(-118.38955358640874, 33.3955978295119));
+        routeCoordinates.add(Point.fromLngLat(-118.389041880506, 33.39578092284221));
+        routeCoordinates.add(Point.fromLngLat(-118.38872797688494, 33.3957916930261));
+        routeCoordinates.add(Point.fromLngLat(-118.38817327048618, 33.39561218978703));
+        routeCoordinates.add(Point.fromLngLat(-118.3872530598711, 33.3956265500598));
+        routeCoordinates.add(Point.fromLngLat(-118.38653065153775, 33.39592811523983));
+        routeCoordinates.add(Point.fromLngLat(-118.38638444985126, 33.39590657490452));
```

Comment:
can't you do a for loop to create those ?

Reply:
This is a list of point taken from the internet: it represents a specific path. I used a for loop for the previous path but it's hard to make a realistic path with a for loop.

Now that the path has been uploaded, I could also consider removing this method. But I thought it was fine because not adding complexity.

Reply:
ok :)